docs/source/scale/distributed.rst

```diff
@@ -0,0 +1,232 @@
+
+For an AWS EC2 cluster, users can also use EFS as the shared file system. Please follow 1) `the instruction of creating EFS <https://docs.aws.amazon.com/efs/latest/ug/gs-step-two-create-efs-resources.html>`_; 2) `the instruction of installing an EFS client <https://docs.aws.amazon.com/efs/latest/ug/installing-amazon-efs-utils.html>`_; and 3) `the instructions of mounting the EFS filesystem <https://docs.aws.amazon.com/efs/latest/ug/efs-mount-helper.html>`_ to set up EFS.
+
+After setting up a shared file system, we can keep all partitioned graph data in the shared folder. Then mount the data folder to the ``/path_to_data/`` of each instances in the cluster so that all GraphStorm containers in the cluster can access these partitioned graph data easily.
+
+Create GraphStorm container by mounting the NFS folder
+.......................................................
+In each instance, use the following command to start a GraphStorm Docker container and run it as a backend daemon.
+
+.. code-block:: shell
+
+    nvidia-docker run -v /path_to_data/:/data \
+                      -v /dev/shm:/dev/shm \
+                      --network=host \
+                      -d --name test graphstorm:local
+
+This command mount the shared ``/path_to_data/`` folder to each container's ``/data/`` folder by which GraphStorm codes can access graph data and save training and inference outcomes.
+
+Setup the IP address file and check port 2222 status
+----------------------------------------------------------
+Collect the IP list
+......................
+The GraphStorm Docker containers use SSH on port ``2222`` to communicate with each other. Users need to collect all IP addresses of the three instances and put them into a text file, e.g., ``/data/ip_list.txt``, which is like:
+
+.. figure:: ../../../tutorial/distributed_ips.png
+    :align: center
+
+.. note:: If possible, use **private IP addresses**, insteand of public IP addresses. Public IP addresses may have additional port constraints, which cause communication issues.
+    
+Put this file into container's ``/data/`` folder.
+
+Check port 2222
+................
+The GraphStorm Docker container uses port ``2222`` to **ssh** to containers running on other machines without passwords. Please make sure all host instances do not use this port.
+
+Users also need to make sure the port ``2222`` is open for **ssh** commands. 
```

Comment:
For distributed training, we also need to make sure ports under 65536 is Open for DistDGL to use.